src/libraries/System.Private.CoreLib/src/Internal/Runtime/CompilerServices/Unsafe.cs

```diff
@@ -420,5 +420,22 @@ public static void SkipInit<T>(out T value)
 
             // ret
         }
+
+        /// <summary>
+        /// Returns a mutable reference to the contents of a boxed struct. The caller should take
+        /// care not to reassign the value except through the struct's own mutable API surface.
+        /// The type of the boxed object is not checked.
```

Comment:
> /// The type of the boxed object is not checked.

It is not checked in this impl, but it is checked in the intrinsic impl...

Reply:
I've updated the comment to be more accurate. I don't think it's all that important to have the two implementations be kept completely in sync, as this is an internal-only API and we can reasonably expect our callers to follow basic hygiene. Do you agree that it's ok to have the two different behaviors here?

Reply:
The idea behind the Internal S.R.CS.Unsafe is that it behaves the same way as the public one.

I do not think that the performance of `Guid.Equals(object o)` really matters. I think it would be ok to keep it as is. If it gets a bit slower as the side-effect of the change, I would not care much. My guess is that the current version of the change makes it slower anyway because of there will be an extra call or jump compared to the baseline.